File: vue-spotify-app.Classes/PendingPlaybackRecordViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace vue_spotify_app.Classes
{
    /// <summary>
    /// View model for storing information about pending playback records.
    /// </summary>
    public class PendingPlaybackRecordViewModel
    {
        public string ID { get; set; }

        public DateTime DateAdded { get; set; }

        public string RecordedTrackName { get; set; }

        public string RecordedSpotifyID { get; set; }

        public string FoundTrackName { get; set; }

        public string FoundAlbumCover { get; set; }

        public List<Artist> FoundArtists { get; set; } = new List<Artist>();

        public string FoundAlbum { get; set; }
    }
}

File: vue-spotify-app.client/src/classes/pendingPlaybackRecordViewModel.ts
import type Artist from "./artist"

export default class PendingPlaybackRecordViewModel{
  id: string = "";
  dateAdded: string = "";
  recordedTrackName: string = "";
  recordedSpotifyID: string = "";
  foundAlbumCover: string = "";
  foundTrackName: string = "";
  foundArtists: string[] = [];
  foundAlbum: string = "";
  isSelected: boolean = false;

  initialise(data){
    this.id = data.id;
    this.dateAdded = data.dateAdded;
    this.recordedTrackName = data.recordedTrackName;
    this.recordedSpotifyID = data.recordedSpotifyID;
    this.foundAlbumCover = data.foundAlbumCover;
    this.foundTrackName = data.foundTrackName;
    this.foundAlbum = data.foundAlbum;
    this.foundArtists = data.foundArtists.map(x => x.name);

  }
}
